src/main/java/com/minecolonies/coremod/commands/ColonyTPCommand.java

```diff
@@ -0,0 +1,176 @@
+            return;
+        }
+
+        EntityPlayer playerToTeleport = null;
+
+        if(sender instanceof EntityPlayer)
+        {
+            playerToTeleport = (EntityPlayer) sender;
+        }
+
+        //If the arguments aren't empty, the sender probably wants to teleport another player.
+        if (args.length != 0 && isPlayerOpped(sender, "ctp"))
+        {
+            final World world = FMLCommonHandler.instance().getMinecraftServerInstance().getEntityWorld();
+            playerToTeleport =
+                    ServerUtils.getPlayerFromUUID(FMLCommonHandler.instance().getMinecraftServerInstance().getPlayerProfileCache()
+                            .getGameProfileForUsername(args[0]).getId(), world);
+
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString("TPin Player: "+ playerToTeleport.getName()));
+        }
+
+        if(playerToTeleport == null)
+        {
+            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(CANT_FIND_PLAYER));
+            return;
+        }
+        playerToTeleport.getCommandSenderEntity().addChatMessage(new TextComponentString("Buckle up buttercup, this ain't no joy ride!!!"));
+        teleportPlayer(sender, playerToTeleport);
+    }
+
+    /**
+     * Method used to teleport the player.
+     * @param sender the sender to have access to the world.
+     * @param playerToTeleport the player which shall be teleported.
+     */
+    private void teleportPlayer(final ICommandSender sender, final EntityPlayer playerToTeleport)
+    {
+        //Now the position will be calculated, we will try up to 4 times to find a save position.
+        int attCounter = 0;
+        while (attCounter <= ATTEMPTS)
+        {
+            attCounter++;
+            /* this math is to get negative numbers */
+            int x = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+                /* keeping X out of the spawn radius */
+                while (x > -SPAWN_NO_TP && x < SPAWN_NO_TP)
+                {
+                    x = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+                }
+
+            /* keeping Z out of the spawn radius */
+            int z = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+                while (z > -SPAWN_NO_TP && z < SPAWN_NO_TP)
+                {
+                    z = rnd.nextInt(UPPER_BOUNDS) - LOWER_BOUNDS;
+                }
+
+                /* Check for a close by colony*/
+            if (ColonyManager.getColony(sender.getEntityWorld(), new BlockPos(x, STARTING_Y, z)) == null)
+            {
+                continue;
+            }
+
+            /*Search for a ground position*/
+            final BlockPos groundPosition = findLand(new BlockPos(x, STARTING_Y, z), sender.getEntityWorld());
+
+            /*If no position found*/
+            if (groundPosition == null)
+            {
+                continue;
+            }
+
+            boolean foundPosition = isPositionSafe(sender, groundPosition);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'foundPosition' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)